### PR TITLE
addHoliday reiview changes  pleaase

### LINKED ISSUE #167: Business tab: Add Holiday (Admin)
### **🏁 Setup**
- [ ] **Assign** yourself to this issue 

### **✅ To do:**
- [ ] **Draw** page mockup using [Whimsical](https://whimsical.com/repairshop-6v7GHoRVFfSSvyvUdeHaQ8) 
(After clicking on the link, you must sign in with your Google Account)

**Frontend**

- [ ] Allow customer to add a holiday
- [ ] Implement frontend

### **🆘 If you need help for the issue you’re working on:**

**1) Write** comments in this issue 
**2) Add** label image
**3) Copy** the issue link and **paste** it to Discord's `#🐛issues` channel
**4)** When solved, **strike** the message in `#🐛issues`, **mark it** as `(SOLVED)`  + **remove** label `help wanted` + **write** comments in this issue 

### *Whenever something is done write a comment below.*

### PR DESCRIPTION
**Pull request by zhannaklimanova**

### CODE REVIEW

**Review by T-Syme — APPROVED** [review]

File: RepairShop-Frontend/src/components/AddHoliday.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        endDate: "",
+        calandarStartDateTime : "", //  bunch of different formats from calandar
+        calandarEndDateTime : "",
+        errorAddHoliday: "",
+        successAddHoliday: ""
+    };
+  },
+  methods: {
+    onSubmit(event) {
+      event.preventDefault();
```
Comment (JDziewonski98) [inline RepairShop-Frontend/src/components/AddHoliday.vue:58]: should add a simple check to make sure end date is after start date
image

File: RepairShop-Frontend/src/components/AddHoliday.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
+
+<template>
+  <div id="addHolidayForm">
```
Comment (JDziewonski98) [inline RepairShop-Frontend/src/components/AddHoliday.vue:3]: Would be good to center the entire page (could wrap everything in a div and play with centering css)
image

**Review by JDziewonski98 — CHANGES_REQUESTED** [review]
add holiday seems good to go other than a minor issue
maybe look at view customer appointments on main to see how to fetch a list of data and populate a table for view holidays

**Review by Shaswata79 — APPROVED** [review]

**Review by sandyl289 — APPROVED** [review]
I agree with Jacek's comments. I saw that you created a blank page for View Holidays. Are you going to create a separate page for "View Holidays"?

- All tests pass (Repository & Service)
- Everything seems to work perfectly  💯 👍🏼 

# Testing
## Adding Holiday (After logging in as Admin)
image

## From Postman (Get All Holidays)
image

File: RepairShop-Frontend/src/components/AddHoliday.vue (hunk 1 above)
Comment (zhannaklimanova) [reply RepairShop-Frontend/src/components/AddHoliday.vue:58]: ok, I'll look into it

**Comment by zhannaklimanova** [comment]
> I agree with Jacek's comments. I saw that you created a blank page for View Holidays. Are you going to create a separate page for "View Holidays"?
> 
> * All tests pass (Repository & Service)
> * Everything seems to work perfectly  💯 👍🏼
> 
> # Testing
> ## Adding Holiday (After logging in as Admin)
> image
> 
> ## From Postman (Get All Holidays)
> image

@sandyl289 Yea, so Tyler and I were talking about it and he said that you're working on  the  display holidays page and that I should move on to viewing/deleting appointments for technicians from the admin account. But If you would like, I can try doing the display holiday thing.

**Comment by sandyl289** [comment]
> > I agree with Jacek's comments. I saw that you created a blank page for View Holidays. Are you going to create a separate page for "View Holidays"?
> > 
> > * All tests pass (Repository & Service)
> > * Everything seems to work perfectly  💯 👍🏼
> > 
> > # Testing
> > ## Adding Holiday (After logging in as Admin)
> > image
> > ## From Postman (Get All Holidays)
> > image
> 
> @sandyl289 Yea, so Tyler and I were talking about it and he said that you're working on the display holidays page and that I should move on to viewing/deleting appointments for technicians from the admin account. But If you would like, I can try doing the display holiday thing.

- The admin can view holidays on the `Holidays: View & Delete` page, so I guess we don't need a separate page now.
image

- As for everyone, they can see the holidays on the `Contact Us` page now 
image

File: RepairShop-Frontend/src/components/TopNavbar.vue [hunk 3]
```diff
@@ -13,10 +13,12 @@
 
     <b-navbar-nav v-show="this.$root.$data.userType == 'Admin'">
       <b-nav-item to="AddService">Add Service</b-nav-item>
+      <b-nav-item to="AddHoliday">Add Holiday</b-nav-item>
     </b-navbar-nav>
 
     <b-navbar-nav>
       <b-nav-item to="ViewServices">View Services</b-nav-item>
+      <b-nav-item to="ViewHolidays">View Holidays</b-nav-item>
```
Comment (sandyl289) [inline RepairShop-Frontend/src/components/TopNavbar.vue]: We can delete this before merging to `main`
```suggestion
```

**Review by sandyl289 — COMMENTED** [review]
- We should remove `ViewHolidays.vue` before merging to `main`
- & remove it from TopNavBar.vue (see [commit suggestion](https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/190#discussion_r604573418) )

# Testing `Add New Holidays` page 

## 1) POSTMAN (See current holidays)
image

## 2) Add New Holiday

### 2A) TEST invalid start date before end date ⚠️
- I agree with Jacek, I think it would be great if we have something to make sure that the `start date` is before the `end date` + error message  ⚠️
image

### 2B) TEST valid dates ✔️
image

##  3) POSTMAN (Make sure holidays were added correctly) ✔️
image

File: RepairShop-Frontend/src/components/AddHoliday.vue (hunk 2 above)
Comment (sandyl289) [reply RepairShop-Frontend/src/components/AddHoliday.vue:3]: Will do this later after merging all branches

File: RepairShop-Frontend/src/components/TopNavbar.vue (hunk 3 above)
Comment (sandyl289) [reply RepairShop-Frontend/src/components/TopNavbar.vue]: done!

**Pull request merged**